src/docfx/build/page/BuildPage.cs

```diff
@@ -203,7 +203,12 @@ private static (List<Error> errors, bool isPage, (OutputMetadata metadata, JObje
         {
             var obj = token as JObject;
 
-            var (schemaValidator, schemaTransformer) = TemplateEngine.GetJsonSchema(file.Mime);
+            if (context.TemplateEngine is null)
+            {
+                throw Errors.TemplateNotDefined(file).ToException();
+            }
+
+            var (schemaValidator, schemaTransformer) = context.TemplateEngine.GetJsonSchema(file.Mime);
```

Comment:
```
var (schemaValidator, schemaTransformer) = context.TemplateEngine?.GetJsonSchema(file.Mime);
```

Reply:
Removed 'TemplateNotDefined', the templateEngine is never null in context.

---
In reply to: [296721036](https://github.com/dotnet/docfx/pull/4789#discussion_r296721036) [](ancestors = 296721036)